### PR TITLE
frontend fixes

### PR DESCRIPTION
**Pull request by ailrst**
- handle unsupported opcodes from https://github.com/UQ-PAC/gtirb-semantics/pull/22
- disambiguate functions with local binding (c `static`) which have the same name but different addresses in the boogie output
- handle relf files with empty relocation table
- some missing relf enum values

### CODE REVIEW

File: src/main/scala/translating/SemanticsLoader.scala [hunk 1]
```diff
@@ -558,6 +569,8 @@ class SemanticsLoader(parserMap: immutable.Map[String, Array[Array[StmtContext]]
       case "__BranchTaken" => None
       case "BTypeNext" => None
       case "BTypeCompatible" => None
+      case "TPIDR_EL0" => Some(Register(name, 64))
```
Comment (l-kent) [inline src/main/scala/translating/SemanticsLoader.scala:572]: What opcodes produce this?

File: src/main/scala/translating/SemanticsLoader.scala [hunk 2]
```diff
@@ -579,15 +592,15 @@ class SemanticsLoader(parserMap: immutable.Map[String, Array[Array[StmtContext]]
...
 
   private def visitIdent(ctx: IdentContext): String = {
     ctx.ID.getText.stripPrefix("\"").stripSuffix("\"")
   }
 
-  private def visitInteger(ctx: IntegerContext): Int = {
-    ctx.DEC.getText.toInt
+  private def visitInteger(ctx: IntegerContext): BigInt = {
+    BigInt(ctx.DEC.getText, 10)
   }
```
Comment (l-kent) [inline src/main/scala/translating/SemanticsLoader.scala:604]: What's the purpose of this change?

File: src/main/scala/util/RunUtils.scala [hunk 3]
```diff
@@ -221,6 +221,13 @@ object IRTransform {
     Logger.debug(
       s"[!] Removed ${before - ctx.program.procedures.size} functions (${ctx.program.procedures.size} remaining)"
     )
+    val dupProcNames = (ctx.program.procedures.groupBy(_.name).filter((n,p) => p.size > 1)).toList.flatMap(_._2)
+
+    var dupCounter = 0 
+    for (p <- dupProcNames) {
+      dupCounter += 1
+      p.name = p.name + "$" + p.address.map(_.toString).getOrElse(dupCounter.toString)
+    }
```
Comment (l-kent) [inline src/main/scala/util/RunUtils.scala:230]: This is good to have but may potentially cause problems elsewhere because there are some things (like indirect call resolution) that rely on the names of procedures in a somewhat fragile way. That is something that I want to fix in the indirect call resolution though.

File: src/main/scala/translating/SemanticsLoader.scala (hunk 2 above)
Comment (ailrst) [reply src/main/scala/translating/SemanticsLoader.scala:604]: The opcode passed through unsupported_op  as an int isn't representible in a signed Int32.

File: src/main/scala/translating/SemanticsLoader.scala (hunk 1 above)
Comment (ailrst) [reply src/main/scala/translating/SemanticsLoader.scala:572]: mrs https://katrinafyi.github.io/aslp-web/5/?opcode=0xd53bd040&bytes=40+D0+3B+D5&asm=mrs+x0%2C+TPIDR_EL0&debug=0

File: src/main/scala/util/RunUtils.scala (hunk 3 above)
Comment (ailrst) [reply src/main/scala/util/RunUtils.scala:230]: Yes its not correct to rely on names it should go through symbol table address -> basil procedure address, we should probably be clear about what the correct unique procedure identifier is in the IR though.

**Review by l-kent — APPROVED** [review]
